File: two-sum/two_sum.ts
function twoSum(nums: number[], target: number): number[] {
  const map = new Map<number, number>()

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i]

    const j = map.get(complement)
    if (map.has(complement) && j !== i){
      return [j, i]
    }

    map.set(nums[i], i)
  }
  return []
}

File: two-sum/two_sum.go
func twoSum(nums []int, target int) []int {
	mp := make(map[int]int)

	for i, num := range nums {
		complement := target - num

		if j, ok := mp[complement]; ok && j != i {
			return []int{i, j}
		}
		mp[num] = i
	}

	return []int{}
}
